workflows_api/runtime/src/collection_publisher.py

```diff
@@ -58,21 +58,25 @@ class Publisher:
 
     def __init__(self) -> None:
         self.func_map = {
-            DataType.zarr: self.create_zarr_collection,
-            DataType.cog: self.create_cog_collection,
+            DataType.zarr.value: self.create_zarr_collection,
+            DataType.cog.value: self.create_cog_collection,
         }
 
     def get_template(self, dataset: Union[ZarrDataset, COGDataset]) -> dict:
         dataset_dict = dataset.dict()
-        collection_dict = {
-            "id": dataset_dict["collection"],
-            **Publisher.common,
-            **{
-                key: dataset_dict[key]
-                for key in Publisher.common_fields
-                if key in dataset_dict.keys()
-            },
-        }
+        collection_dict = dataset_dict
+        # manage data as needed by ingest
+        collection_dict["id"] = dataset_dict["collection"]
+        collection_dict.update(Publisher.common)
+        collection_dict.update({
+            key: dataset_dict[key]
+            for key in Publisher.common_fields
+            if key in dataset_dict.keys()
+        })
+        # Convert enum values to their string equivalents
+        for key, value in dataset_dict.items():
+            if isinstance(value, DataType):
+                dataset_dict[key] = value.value
```

Comment:
there was a key value namely `datatype: <Datatype.cog>`
It was not a valid json for the ingest api call

Reply:
The value for the key `data_type` in the collection is an enumerator. Its value will be `<DataType.cog>` by default, which is not a valid JSON and will cause an issue during API call for ingest. Previously, this was ignored in the JSON body for ingest, However, as we will allow everything now, this will come in as well. So fixed it here.